File: client/management/types/order-dto.ts
import { CreateItemDto, ItemDto } from "./item-dto";

export interface OrderDto {
  orderCode: string;
  orderDate: string;
  maintenanceDate: string;
  price: number;
  orderItems: ItemDto[];
}
export interface CreateCustomerOrderCommandRequest {
  totalPrice: number;
  maintenanceDay: number;
  orderItems: CreateItemDto[];
}

export interface CreateSupplierOrderCommandRequest {
  supplierId: string;
  totalPrice: number;
  orderItems: CreateItemDto[];
}

File: server/CRMApp/src/CRMApp.Application/Features/Dtos/Order/OrderDto.cs
using CRMApp.Application.Features.Dtos.Item;
using CRMApp.Domain.Enums.Order;

namespace CRMApp.Application.Features.Dtos.Order
{
    public record OrderDto(string OrderCode, DateTime OrderDate, DateTime? MaintenanceDate, decimal Price, List<ItemDto> OrderItems);
}
